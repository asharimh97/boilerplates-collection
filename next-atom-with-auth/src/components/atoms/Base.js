import styled from "styled-components";
import {
  compose,
  layout,
  position,
  space,
  color,
  flexbox,
  grid,
  background,
  border,
  shadow,
  typography
} from "styled-system";
import propTypes from "@styled-system/prop-types";

import theme from "styles/theme";

const Base = styled("div")(
  compose(
    background,
    border,
    color,
    flexbox,
    grid,
    layout,
    position,
    shadow,
    space,
    typography
  )
);

Base.displayName = "Base";
Base.defaultProps = {
  theme
};

Base.propTypes = {
  ...propTypes.background,
  ...propTypes.border,
  ...propTypes.color,
  ...propTypes.flexbox,
  ...propTypes.grid,
  ...propTypes.layout,
  ...propTypes.position,
  ...propTypes.shadow,
  ...propTypes.space,
  ...propTypes.typography
};

export default Base;
