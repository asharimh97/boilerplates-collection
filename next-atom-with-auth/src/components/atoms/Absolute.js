import styled from "styled-components";
import Base from "./Base";

const Absolute = styled(Base)`
  position: absolute;
`;

export default Absolute;
