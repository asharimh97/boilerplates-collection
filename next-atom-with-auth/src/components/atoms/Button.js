import { Button as AntButton } from "antd";
import PropTypes from "prop-types";
import styled, { css } from "styled-components";
import { flexbox, layout, space, typography } from "styled-system";
import propTypes from "@styled-system/prop-types";

const flex = props => props.flex && { display: "flex" };

const color = props => {
  if (props.color) {
    if (props.type === "link" || props.type === "text") {
      return css`
        &,
        &:hover,
        &:active,
        &:focus {
          color: ${props.theme.colors[props.color]?.["primary"] || props.color};
        }
      `;
    }

    return css`
      background-color: ${props.theme.colors[props.color]?.["primary"] ||
      props.color};
      border-color: ${props.theme.colors[props.color]?.["primary"] ||
      props.color};
      color: white;

      &:hover,
      &:active,
      &:focus {
        background-color: ${props.theme.colors[props.color]?.["60"] ||
        props.color};
        border-color: ${props.theme.colors[props.color]?.["60"] || props.color};
        color: white;
      }
    `;
  }

  return null;
};

const Button = styled(AntButton)`
  ${color};

  ${flex};
  ${flexbox};
  ${layout};
  ${space};
  ${typography}
`;

Button.propTypes = {
  flex: PropTypes.bool,
  ...propTypes.flexbox,
  ...propTypes.layout,
  ...propTypes.space,
  ...propTypes.typography
};

Button.displayName = "Button";

export default Button;
