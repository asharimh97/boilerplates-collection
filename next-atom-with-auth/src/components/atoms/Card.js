import styled from "styled-components";
import Base from "./Base";

const Card = styled(Base)``;

Card.defaultProps = {
  background: "white",
  p: 3,
  borderRadius: "5px",
  boxShadow: 5
};

Card.displayName = "Card";

export default Card;
