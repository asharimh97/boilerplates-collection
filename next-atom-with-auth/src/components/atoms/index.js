export { default as Absolute } from "./Absolute";
export { default as Avatar } from "./Avatar";
export { default as Box } from "./Box";
export { default as Button } from "./Button";
export { default as Card } from "./Card";
export { default as Container } from "./Container";
export { default as Divider } from "./Divider";
export { default as Fixed } from "./Fixed";
export { default as Flex } from "./Flex";
export { default as Grid } from "./Grid";
export { default as Image } from "./Image";
export { default as Link } from "./Link";
export { default as Sticky } from "./Sticky";
export { default as Text } from "./Text";
export { default as Title } from "./Title";
