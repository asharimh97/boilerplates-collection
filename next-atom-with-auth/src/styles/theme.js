const createMediaQuery = n => `@media screen and (min-width:${n})`;

const addAliases = (arr, aliases) =>
  aliases.forEach((key, i) =>
    Object.defineProperty(arr, key, {
      enumerable: false,
      get() {
        return this[i];
      }
    })
  );

export const breakpoints = ["426px", "768px", "992px", "1200px"];

export const mediaQueries = breakpoints.map(createMediaQuery);

const aliases = ["sm", "md", "lg", "xl"];

addAliases(breakpoints, aliases);
addAliases(mediaQueries, aliases);
export const font = "Google Sans, Helvetica, Arial, sans-serif";

// this font size scaling follows styled-system standard
// however we can't change the standard scaling, theme will always return
// this sequence `[12, 14, 16, 20, 24, 32, 48]` even if we try to shift
// some values between the scales.
export const fontSizes = [12, 14, 16, 20, 24, 32, 48];
fontSizes.midSmall = 18;
fontSizes.mid = 28;
fontSizes.midBig = 40;

export const light = 300;
export const regular = 400;
export const medium = 500;
export const bold = 700;
export const fontWeights = {
  light,
  regular,
  medium,
  bold
};

// in this new spacing standard index `0` stands for 0 margin/padding
export const space = [0, 4, 8, 16, 24, 32, 40, 64, 128, 256, 512];
const shades = {
  black: {
    100: "#25282B",
    80: "#52575C",
    60: "#A0A4A8",
    40: "#CACCCF",
    20: "#EBEBEB",
    10: "#F5F5F5",
    5: "#FAFAFA",
    primary: "#25282B"
  },
  red: {
    80: "#99231D",
    60: "#cc2f26",
    50: "#ff3b30",
    40: "#ff8983",
    10: "#ffd8d6",
    primary: "#ff3b30"
  },
  green: {
    80: "#007a00",
    60: "#009b00",
    50: "#00b900",
    40: "#66d566",
    10: "#e5f8e5",
    primary: "#00b900"
  },
  orange: {
    80: "#c26307",
    60: "#e07b03",
    50: "#ff9500",
    40: "#ffc066",
    10: "#ffeacc",
    primary: "#ff9500"
  },
  blue: {
    80: "#02577e",
    60: "#039de3",
    50: "#1cb7fc",
    40: "#4fc7fd",
    10: "#e6f7ff",
    primary: "#1cb7fc"
  }
};

const textColor = {
  primary: shades.black[100],
  secondary: shades.black[80],
  tertiary: shades.black[60],
  quarternary: shades.black[40]
};

const inputColors = {
  success: shades.green,
  danger: shades.red,
  warning: shades.orange,
  disabled: shades.black,
  info: shades.blue
};

export const colors = {
  ...shades,
  ...inputColors,
  textColor,
  primary: shades.orange
};

export const radius = "3px";
export const shadows = [
  `0 1px 1px 0 rgba(10, 31, 68, 0.08), 0 0 1px 0 rgba(10, 31, 68, 0.08)`,
  `0 3px 4px 0 rgba(10, 31, 68, 0.1), 0 0 1px 0 rgba(10, 31, 68, 0.08)`,
  `0 8px 10px 0 rgba(10, 31, 68, 0.1), 0 0 1px 0 rgba(10, 31, 68, 0.08)`,
  `0 18px 18px 0 rgba(10, 31, 68, 0.12), 0 0 1px 0 rgba(10, 31, 68, 0.1)`,
  `0 26px 26px 0 rgba(10, 31, 68, 0.12), 0 0 1px 0 rgba(10, 31, 68, 0.1)`,
  `0 3px 32px 0 rgba(10, 31, 68, 0.1), 0 0 1px 0 rgba(10, 31, 68, 0.08)`
];

export const radii = ["3px", "5px"];
export const lineHeights = ["1", "1.5"];

export default {
  bold,
  breakpoints,
  colors,
  font,
  fontSizes,
  fontWeights,
  light,
  lineHeights,
  mediaQueries,
  medium,
  radii,
  radius,
  regular,
  shadows,
  space
};
