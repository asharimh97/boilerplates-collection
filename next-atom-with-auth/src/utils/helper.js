/**
 * Add fallback to placeholder image
 * @param {Element} e - HTML Element target
 */
export const handleErrorImage = e => {
  e.target.onerror = null;
  e.target.src = "/static/images/placeholder.png";
};

export const convertNumberToPixel = size => {
  if (Number.isInteger(size)) return `${size}px`;

  return size;
};
