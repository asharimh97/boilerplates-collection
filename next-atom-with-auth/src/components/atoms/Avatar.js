import styled, { css } from "styled-components";
import { convertNumberToPixel } from "utils/helper";
import Image from "./Image";

const sizes = {
  small: { width: "40px", height: "40px" },
  regular: { width: "50px", height: "50px" },
  big: { width: "65px", height: "65px" }
};

const Avatar = styled(Image).attrs(props => ({
  autofit: true,
  height: sizes[props.size]?.width || convertNumberToPixel(props.size),
  fullRounded: props.circle
}))`
  ${props =>
    props.size && !Number.isInteger(props.size)
      ? css`
          ${sizes[props.size]}
        `
      : css`
          width: ${props.size};
          height: ${props.size};
        `}
`;

Avatar.propTypes = {};

Avatar.displayName = "Avatar";

Avatar.defaultProps = {
  size: "regular",
  src: "/static/images/avatar.jpeg"
};

export default Avatar;
