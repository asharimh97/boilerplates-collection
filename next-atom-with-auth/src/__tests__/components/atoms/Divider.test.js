import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import { create } from "react-test-renderer";
import "jest-styled-components";

import { Divider } from "components/atoms";
import { convertNumberToPixel } from "utils/helper";
import { space } from "styles/theme.base";

test("renders `Divider` without errors", () => {
  const { getByTestId } = render(<Divider data-testid="divider-component" />);
  expect(getByTestId("divider-component")).toBeInTheDocument();
});
test("renders `Divider` with additional props", () => {
  const divider = create(<Divider my={4} />).toJSON();

  expect(divider).toMatchSnapshot();
  expect(divider).toHaveStyleRule("margin-top", convertNumberToPixel(space[4]));
  expect(divider).toHaveStyleRule(
    "margin-bottom",
    convertNumberToPixel(space[4])
  );
});
test("renders `Divider` with `vertical` prop", () => {
  const divider = create(<Divider vertical mx={3} />).toJSON();

  expect(divider).toMatchSnapshot();
  expect(divider).toHaveStyleRule("width", "1px");
  expect(divider).toHaveStyleRule("display", "inline-block");
  expect(divider).toHaveStyleRule(
    "margin-right",
    convertNumberToPixel(space[3])
  );
  expect(divider).toHaveStyleRule(
    "margin-left",
    convertNumberToPixel(space[3])
  );
});
