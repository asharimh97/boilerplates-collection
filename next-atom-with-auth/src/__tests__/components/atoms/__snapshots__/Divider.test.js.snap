// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders \`Divider\` with \`vertical\` prop 1`] = `
.c0 {
  margin-left: 16px;
  margin-right: 16px;
  border-color: #ccc;
}

.c1 {
  border: none;
  border-top: solid 1px;
  color: transparent;
  display: block;
  width: 100%;
  display: inline-block;
  border-top: none;
  border-left: solid 1px;
  font-size: inherit;
  margin: 0 12px;
  min-height: 0.9em;
  width: 1px;
  margin-left: 16px;
  margin-right: 16px;
  border-color: #ccc;
}

<div
  className="c0 c1"
/>
`;

exports[`renders \`Divider\` with additional props 1`] = `
.c0 {
  margin-top: 24px;
  margin-bottom: 24px;
  border-color: #ccc;
}

.c1 {
  border: none;
  border-top: solid 1px;
  color: transparent;
  display: block;
  width: 100%;
  border-color: #ccc;
}

<div
  className="c0 c1"
/>
`;
